### DISCUSSION TITLE
Is there a way to change the layer position of a component programmatically?

### DISCUSSION THREAD

**Discussion by bimsina** [GrapesJS/grapesjs#4541] [Q&A] [answered]
I am trying to implement a feature where a selected component would be moved behind, but I am trying to achieve that programmatically (using a context menu) instead of dragging the component in the LayerManager UI.
<img width="450" alt="Screen Shot 2022-08-30 at 6 47 50 PM" src="https://user-images.githubusercontent.com/29589003/187443731-60f29332-4a12-4a93-8856-1c2b956fb4a2.png">

So far I've tried to set the zIndex of the selected component using this

`editor.getSelected()?.setStyle({ zIndex: -10 });`

and also with this

`editor.getSelectedToStyle().setStyle({ zIndex: -10 });`

but no changes are visible.

Any help would be appreciated 🙏 .

**Comment by artf** [comment, marked as answer]
I'd suggest using the StyleManager API in order to update properly the selected style rule.
```js
editor.StyleManager.getSelected()?.addStyle({ 'z-index': -10 })
```

**Reply by bimsina** [reply]
Looks like this works as well

`editor.getSelected()?.setStyle({ 'z-index': -10 });
`

Thanks for your help  🙏

**Reply by badiuciprian** [reply]
yes, but then you lose the synchronization with the layer manager
